engine/src/test/java/io/zeebe/engine/processing/job/ActivateJobsTest.java

```diff
@@ -111,16 +111,49 @@ public void shouldRejectInvalidType() {
   }
 
   @Test
-  public void shouldRejectInvalidWorker() {
+  public void shouldNotRejectEmptyWorker() {
+    // given
+    ENGINE.deployment().withXmlResource(PROCESS_ID, MODEL_SUPPLIER.apply(taskType)).deploy();
+    final long firstInstanceKey = createWorkflowInstances(3, "{'foo':'bar'}").get(0);
+
+    final long expectedJobKey =
+        jobRecords(JobIntent.CREATED)
+            .withType(taskType)
+            .filter(r -> r.getValue().getWorkflowInstanceKey() == firstInstanceKey)
+            .getFirst()
+            .getKey();
+
+    final Duration timeout = Duration.ofMinutes(12);
+
     // when
     final Record<JobBatchRecordValue> batchRecord =
-        ENGINE.jobs().withType(taskType).byWorker("").expectRejection().activate();
+        ENGINE
+            .jobs()
+            .withType(taskType)
+            .withTimeout(timeout.toMillis())
+            .withMaxJobsToActivate(1)
+            .activate();
+
+    final List<JobRecordValue> jobs = batchRecord.getValue().getJobs();
+    final List<Long> jobKeys = batchRecord.getValue().getJobKeys();
 
     // then
-    assertThat(batchRecord)
-        .hasRejectionType(RejectionType.INVALID_ARGUMENT)
-        .hasRejectionReason(
-            "Expected to activate job batch with worker to be present, but it was blank");
+    assertThat(batchRecord.getIntent()).isEqualTo(JobBatchIntent.ACTIVATED);
+
+    assertThat(jobKeys).hasSize(1);
```

Comment:
I think it's fine to not test these things - we already have tests covering this. Rule of thumb is to have many tests, each asserting as little as possible (the ideal is one assertion per test, but obviously that's not always possible). So in this case, what is our test covering? What are the things that we really need to assert?

I would argue only care that one job was activated, and that its worker is nothing. The rest isn't important.

Reply:
I agree, will fix that. By the way, what do you think about parameterized tests? We can simply reuse some tests that already exist, but with different worker name.

Reply:
I decided to not use parameterized tests in that case. I removed unnecessary parts.

Reply:
We'll be refactoring parts of the engine and how we process job, so I imagine some of the tests will change - best not to modify too much now :slightly_smiling_face: